src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java

```diff
@@ -1194,10 +1194,15 @@ public Response getGuestbookResponsesByDataverse(@Context ContainerRequestContex
             public void write(OutputStream os) throws IOException,
                     WebApplicationException {
 
-                Dataverse dv = dataverseService.findByAlias(dvIdtf);
+                Dataverse dv;
+                try {
+                    dv = findDataverseOrDie(dvIdtf);
+                } catch (WrappedResponse wr) {
+                    throw new WebApplicationException(wr.getResponse());
```

Comment:
```suggestion
                    return wr.getResponse());
```

Untested. Can you please give this a try? It's our normal pattern.

Reply:
Indeed, I have seen this pattern before (e.g. on line 1188). But void methods cannot return a value, the IDE helpfully says 😄 

My gut tells me this lookup should never fail, because the same lookup is performed a few lines up. Unfortunately I couldn't find a way to reference the Dataverse object from within this `write` method to circumvent the whole lookup with try-catch pattern alltogether.

Reply:
Ah, now that I'm in front of my IDE I see what you mean. 😄 

I have the same gut feeling but how to do you feel about writing some API tests? I'd suggest adding a method for downloading guestbook responses to UtilIT.java (I couldn't find one) and then calling it from DataversesIT.java.

Reply:
> couldn't find a way to reference the Dataverse object

My IDE doesn't complain about this:

```
diff --git a/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java b/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java
index 30c1453525..3fd17c517b 100644
--- a/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java
+++ b/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java
@@ -1173,8 +1173,9 @@ public class Dataverses extends AbstractApiBean {
     public Response getGuestbookResponsesByDataverse(@Context ContainerRequestContext crc, @PathParam("identifier") String dvIdtf,
             @QueryParam("guestbookId") Long gbId, @Context HttpServletResponse response) {
 
+        Dataverse dv;
         try {
-            Dataverse dv = findDataverseOrDie(dvIdtf);
+            dv = findDataverseOrDie(dvIdtf);
             User u = getRequestUser(crc);
             DataverseRequest req = createDataverseRequest(u);
             if (permissionSvc.request(req)
@@ -1194,7 +1195,6 @@ public class Dataverses extends AbstractApiBean {
             public void write(OutputStream os) throws IOException,
                     WebApplicationException {
 
-                Dataverse dv = dataverseService.findByAlias(dvIdtf);
                 Map<Integer, Object> customQandAs = guestbookResponseService.mapCustomQuestionAnswersAsStrings(dv.getId(), gbId);
                 Map<Integer, String> datasetTitles = guestbookResponseService.mapDatasetTitles(dv.getId());
```

Reply:
> My IDE doesn't complain about this

Mine doesn't either! Nice!

Reply:
> how to do you feel about writing some API tests?

Writing API tests is a bit too much at this moment, sorry. Maybe later this month.